### PR TITLE
Issue #357: refact contributing docs

### LINKED ISSUE #357: Update contributing docs
# Issue description

We need to refactor the whole contributing docs to keep the essential requirements and making onboarding easier.

## Type of Issue

- [ ] Documentation update

### PR DESCRIPTION
**Pull request by mathpluscode**
# Description

Simplify the contributing docs to keep essential components.

Fixes #357

## Type of change

What types of changes does your code introduce to DeepReg? _Put an `x` in the boxes that
apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Documentation Update (fix or improvement on the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [x] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [ ] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mathpluscode** [comment]
@agrimwood  @zacbaum @ebonmati 

I'm trying to simplify the contributing guideline, the current one is too nested and people actually ignore the long pages. Also some guidelines like commit messages, it's not possible to test them, so we can simply saying commit messages should be meaningful (if you want we can add 1 or 2 examples, but 10 examples might be overkill).

Would you have time to review this?

**Comment by zacbaum** [comment]
I can take a look at this sometime this week, will assign myself

**Comment by agrimwood** [comment]
Thanks for taking this on, definitely worth it. Will take a look on Thursday.

File: docs/source/contributing/guide.md [hunk 1]
```diff
@@ -1,11 +1,61 @@
...
+   [testing guidelines](test.html) for further details.
+
+   c. Please write meaningful docstring and documentations for added or changed code.
+
+   We use
+   [Sphinx docstring format](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html).
+
+   d. Please update the
+   [CHANGELOG](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md) regarding
+   the changes.
```
Comment (zacbaum) [inline docs/source/contributing/guide.md:49]: The points under a/b/c don't display all that nicely (or obviously) as subpoints in the RTD. Is it possible to change that?

See attached photo. Can we indent the bits marked in red?

image

**Review by zacbaum — CHANGES_REQUESTED** [review]
See changes in comments - only minor.

@mathpluscode - I may as well update the release guidelines now? ~Will get to that this evening and commit to this ticket so we can get those out of the way~ _(edit: Done!)_ - sorry for dragging on that... 😅

File: docs/source/contributing/guide.md (hunk 1 above)
Comment (mathpluscode) [reply docs/source/contributing/guide.md:49]: ooops will add more space ;)

**Comment by mathpluscode** [comment]
> See changes in comments - only minor.
> 
> @mathpluscode - I may as well update the release guidelines now? ~Will get to that this evening and commit to this ticket so we can get those out of the way~ _(edit: Done!)_ - sorry for dragging on that... 😅

Feel free to push any commit you want!

**Comment by zacbaum** [comment]
> > See changes in comments - only minor.
> > @mathpluscode - I may as well update the release guidelines now? ~Will get to that this evening and commit to this ticket so we can get those out of the way~ _(edit: Done!)_ - sorry for dragging on that... 😅
> 
> Feel free to push any commit you want!

It's in :) Make any changes or ask for clarification if something is unclear.

File: docs/source/contributing/guide.md [hunk 2]
```diff
@@ -1,11 +1,61 @@
 # Guideline
```
Comment (agrimwood) [inline docs/source/contributing/guide.md:1]: Should change to Guidelines

File: docs/source/contributing/guide.md [hunk 3]
```diff
@@ -1,11 +1,61 @@
...
-
-- For reporting bugs, or to request features, please
-  [raise an issue](https://github.com/DeepRegNet/DeepReg/issues/new/choose).
-- For fixing bugs, or implementing features, please
-  [send a pull request](pull_request.html).
-- For adding DeepReg Demos, please check the additional requirements in
-  [Add a DeepReg Demo](demo.html).
+We welcome contributions to DeepReg. Please
+[raise an issue](https://github.com/DeepRegNet/DeepReg/issues/new/choose) to report
+bugs, request features, or asking questions. For code contribution, please follow the
```
Comment (agrimwood) [inline docs/source/contributing/guide.md:5]: 'ask questions' - another minor comment

**Review by agrimwood — APPROVED** [review]
Minor feedback in comments, happy to approve PR regardless. @zacbaum spotted the main issue , which was list formatting.

File: docs/source/contributing/guide.md (hunk 3 above)
Comment (mathpluscode) [reply docs/source/contributing/guide.md]: solved :)

File: docs/source/contributing/guide.md (hunk 2 above)
Comment (mathpluscode) [reply docs/source/contributing/guide.md]: solved :)

File: docs/source/contributing/guide.md (hunk 1 above)
Comment (mathpluscode) [reply docs/source/contributing/guide.md]: solved :)

**Comment by mathpluscode** [comment]
@zacbaum plz have a look now https://deepreg--548.org.readthedocs.build/en/548/contributing/guide.html, if ok plz approve :)

**Comment by zacbaum** [comment]
> @zacbaum plz have a look now https://deepreg--548.org.readthedocs.build/en/548/contributing/guide.html, if ok plz approve :)

Numbering is a bit confusing at first glance, but much better than unindented 😄

**Review by zacbaum — APPROVED** [review]

**Comment by mathpluscode** [comment]
> > @zacbaum plz have a look now https://deepreg--548.org.readthedocs.build/en/548/contributing/guide.html, if ok plz approve :)
> 
> Numbering is a bit confusing at first glance, but much better than unindented 😄

:( locally in my IDE it was i, ii, iii, anyway next time we try bullet points

**Pull request merged**